[Bug] StaticLoader name collision compilation error

I updated my project to use `leptos-fluent` v0.1.17 and `fluent-templates` v0.10.0. When I try to compile my project, I get this error:

```
error[E0308]: mismatched types
  --> src/lib.rs:50:3
   |
50 | /   leptos_fluent! {{
51 | |     translations: [TRANSLATIONS],
52 | |     locales: "./locales",
53 | |     check_translations: "./src/**/*.rs",
...  |
59 | |     initial_language_from_navigator: true
60 | |   }};
   | |____^ expected `StaticLoader`, found a different `StaticLoader`
   |
   = note: `StaticLoader` and `StaticLoader` have similar names, but are actually distinct types
note: `StaticLoader` is defined in crate `fluent_templates`
  --> /home/insomnia/.cargo/registry/src/index.crates.io-6f17d22bba15001f/fluent-templates-0.10.0/src/loader/static_loader.rs:12:1
   |
12 | pub struct StaticLoader {
   | ^^^^^^^^^^^^^^^^^^^^^^^
note: `StaticLoader` is defined in crate `fluent_templates`
  --> /home/insomnia/.cargo/registry/src/index.crates.io-6f17d22bba15001f/fluent-templates-0.9.4/src/loader/static_loader.rs:12:1
   |
12 | pub struct StaticLoader {
   | ^^^^^^^^^^^^^^^^^^^^^^^
   = note: perhaps two different versions of crate `fluent_templates` are being used?
   = note: this error originates in the macro `leptos_fluent` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
warning: `lemmy-ui-leptos` (lib) generated 14 warnings
error: could not compile `lemmy-ui-leptos` (lib) due to 1 previous error; 14 warnings emitted
Error: Failed to build lemmy-ui-leptos
```

The collision appears to be happening because it's trying to look both for the `fluent-templates` 0.9.4 `StaticLoader` and 0.10.0 `StaticLoader`. A look through my `Cargo.lock` file reveals the following:

```
...
[[package]]
name = "fluent-template-macros"
version = "0.9.4"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "77d2bcae1f3ec390c50161fcf130d3228750e9ecf965618584e046d884199b83"
dependencies = [
 "flume",
 "ignore",
 "once_cell",
 "proc-macro2",
 "quote",
 "syn 2.0.75",
 "unic-langid",
]
...
[[package]]
name = "fluent-templates"
version = "0.9.4"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "197feb1e37209c6b3d29f0754b11fc070890efb2b1d761caac4e5287a200e9db"
dependencies = [
 "arc-swap",
 "fluent",
 "fluent-bundle",
 "fluent-langneg",
 "fluent-syntax",
 "fluent-template-macros 0.9.4",
 "flume",
 "heck",
 "ignore",
 "intl-memoizer",
 "log",
 "once_cell",
 "serde_json",
 "thiserror",
 "unic-langid",
]
...
[[package]]
name = "leptos-fluent"
version = "0.1.17"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "ed6f0c7684f034decab4bfe92291c57e82ab27dd5aa3beef325c8d5f3ce42e63"
dependencies = [
 "fluent-templates 0.9.4",
 "leptos",
 "leptos-fluent-macros",
 "leptos_meta",
 "web-sys",
]

[[package]]
name = "leptos-fluent-macros"
version = "0.1.17"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "347676182d2064b95924f063a31ffac63ab3836e8d619adc722ea9bff367a37b"
dependencies = [
 "cfg-expr",
 "current_platform",
 "fluent-syntax",
 "fluent-templates 0.9.4",
 "globwalk",
 "pathdiff",
 "proc-macro2",
 "quote",
 "serde",
 "serde_json",
 "syn 2.0.75",
 "walkdir",
]
```

An examination of `leptos-fluent`'s codebase doesn't show any use of v0.9.4 anywhere - all of the uses are for v0.10.0. I am fortunately able to get my project to compile if I knock my `fluent-template` version down to 0.9.4, but I imagine you want to have users of this library use the latest version of `fluent-templates` when possible. To this end, it may be worth re-exporting the `fluent-templates` crate from this crate so users don't have to add the dependencies separately and so it's easier to guarantee that dependents use compatible versions.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # CHANGELOG
 
+## 2024-08-18 - [0.1.18]
+
+- Relax `fluent-templates` dependency.
+
 ## 2024-08-17 - [0.1.17]
 
 ### New features
@@ -480,6 +484,7 @@ version to `0.1` during installation.
 
 - Added all ISO-639-1 and ISO-639-2 languages.
 
+[0.1.18]: https://github.com/mondeja/leptos-fluent/compare/v0.1.17...v0.1.18
 [0.1.17]: https://github.com/mondeja/leptos-fluent/compare/v0.1.16...v0.1.17
 [0.1.16]: https://github.com/mondeja/leptos-fluent/compare/v0.1.15...v0.1.16
 [0.1.15]: https://github.com/mondeja/leptos-fluent/compare/v0.1.14...v0.1.15

--- a/leptos-fluent-macros/Cargo.toml
+++ b/leptos-fluent-macros/Cargo.toml
@@ -2,7 +2,7 @@
 name = "leptos-fluent-macros"
 description = "Macros for leptos-fluent"
 edition.workspace = true
-version = "0.1.17"
+version = "0.1.18"
 license = "MIT"
 documentation.workspace = true
 repository.workspace = true
@@ -20,7 +20,7 @@ quote = "1"
 syn = { version = "2", features = ["visit", "full"] }
 walkdir = ">=2"
 globwalk = "0.9"
-fluent-templates = { version = "0.10", default-features = false, features = [
+fluent-templates = { version = ">=0.9", default-features = false, features = [
   "macros",
   "walkdir"
 ] }

--- a/leptos-fluent/Cargo.toml
+++ b/leptos-fluent/Cargo.toml
@@ -2,7 +2,7 @@
 name = "leptos-fluent"
 description = "Fluent framework for internationalization of Leptos applications"
 edition.workspace = true
-version = "0.1.17"
+version = "0.1.18"
 license = "MIT"
 documentation.workspace = true
 repository.workspace = true
@@ -12,7 +12,7 @@ keywords.workspace = true
 
 [dependencies]
 leptos-fluent-macros = { path = "../leptos-fluent-macros" }
-fluent-templates = { version = "0.10", default-features = false, features = [
+fluent-templates = { version = ">=0.9", default-features = false, features = [
   "macros",
   "walkdir"
 ] }

